internal/storagenode/logstream/executor_test.go

```diff
@@ -1380,14 +1383,523 @@ func TestExecutor_Recover(t *testing.T) {
+				require.Zero(t, lhwm)
+
+				rpt, err := lse.Report(context.Background())
+				require.NoError(t, err)
+				require.Equal(t, snpb.LogStreamUncommitReport{
+					LogStreamID:           lsid,
+					UncommittedLLSNOffset: 2,
+					UncommittedLLSNLength: 0,
+					Version:               1,
+					HighWatermark:         1,
+				}, rpt)
+
+				lsrmd, err := lse.Metadata()
+				require.NoError(t, err)
+				require.Equal(t, varlogpb.LogStreamStatusSealed, lsrmd.Status)
+				require.Equal(t, types.Version(1), lsrmd.Version)
+				require.Equal(t, types.GLSN(1), lsrmd.GlobalHighWatermark)
+				// NOTE: Watermark values are invalid because all log entries are already trimmed.
+				require.Equal(t, varlogpb.LogSequenceNumber{LLSN: 0, GLSN: 0}, lsrmd.LocalLowWatermark)
+				require.Equal(t, varlogpb.LogSequenceNumber{LLSN: 0, GLSN: 0}, lsrmd.LocalHighWatermark)
+			},
+		},
+		{
+			// A log stream replica with no log entries due to trim but the
+			// commit context indicating GLSN 1 as the local high watermark
+			// should become state "sealing" by invoking Seal RPC with GLSN 2
+			// as the last committed GLSN.
+			//
+			// logs    : _ (empty)
+			// cc      : 1
+			// seal    : 2
+			// expected: sealing
+			name: "LocalHighWatermarkZero_CommitContextOne_LastCommitTwo",
+			pref: func(t *testing.T, lse *Executor) {
+				storage.TestSetCommitContext(t, lse.stg, storage.CommitContext{
+					Version:            1,
+					HighWatermark:      1,
+					CommittedGLSNBegin: 1,
+					CommittedGLSNEnd:   2,
+					CommittedLLSNBegin: 1,
+				})
+			},
+			postf: func(t *testing.T, lse *Executor) {
+				lss, lhwm, err := lse.Seal(context.Background(), 2)
+				require.NoError(t, err)
+				require.Equal(t, varlogpb.LogStreamStatusSealing, lss)
+				require.Zero(t, lhwm)
+
+				rpt, err := lse.Report(context.Background())
+				require.NoError(t, err)
+				require.Equal(t, snpb.LogStreamUncommitReport{
+					LogStreamID:           lsid,
+					UncommittedLLSNOffset: 2,
+					UncommittedLLSNLength: 0,
+					Version:               1,
+					HighWatermark:         1,
+				}, rpt)
+
+				lsrmd, err := lse.Metadata()
+				require.NoError(t, err)
+				require.Equal(t, varlogpb.LogStreamStatusSealing, lsrmd.Status)
+				require.Equal(t, types.Version(1), lsrmd.Version)
+				require.Equal(t, types.GLSN(1), lsrmd.GlobalHighWatermark)
+				// NOTE: Watermark values are invalid because all log entries are already trimmed.
+				require.Equal(t, varlogpb.LogSequenceNumber{LLSN: 0, GLSN: 0}, lsrmd.LocalLowWatermark)
+				require.Equal(t, varlogpb.LogSequenceNumber{LLSN: 0, GLSN: 0}, lsrmd.LocalHighWatermark)
+			},
+		},
+		{
+			// A log stream replica, with GLSN 2 as the local high watermark
+			// and the commit context indicating GLSN 2 as the local high
+			// watermark, should become state "sealing" by invoking Seal RPC
+			// with GLSN 1 as the last committed GLSN.
+			//
+			// logs    : 1 2
+			// cc      :   2
+			// seal    : 1
+			// expected: sealing
```

Comment:
I don't understand why you expect a `sealing` state.

Reply:
As you see, it currently panics. Panic will result in shutting down a storage node, leading termination of all log stream replicas in the storage node. In addition, only operators can fix it correctly.
Therefore, I want the log stream replica to be SEALING rather than panic. Transiting the status of the log stream replica to SEALING won't shut down the storage node. We need a particular method to fix it automatically, such as removing all logs and synchronizing between replicas.

Reply:
It should not happen, so I am not sure when it can happen. This is why it panics; for debugging.
I am not considering an invalid seal request; the metadata repository always tells the truth.

Reply:
Ok, but I still don't know what to fix it.